pkg/reconciler/config.go

```diff
@@ -21,13 +21,21 @@ import (
 
 	"github.com/fsnotify/fsnotify"
 	"github.com/spf13/viper"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 )
 
 // GlobalConfig is the configuration for the controllers, it is
 // supposed to be populated from the configmap attached to the
 // controller manager.
 type GlobalConfig struct {
-	ISBSvc *ISBSvcConfig `json:"isbsvc"`
+	StandardResources *ResourceConfig `json:"standardResources"`
```

Comment:
```
data:
  controller-config.yaml: |+
    defaults:
      containerResources:
          limits:
            cpu: xxxx
            memory: xxxx
          requests:
            cpu: xxx
            memory: xxxx
    isbsvc:
      redis:
...
```

This way we may add more `defaults` in the future.

Also, use `*apiv1.ResourceRequirements` directly.

Reply:
Agree with additional defaults in the future. ResourceList is a Map which makes the valid yaml as following, if I were to use `*apiv1.ResourceRequirements` directly.

```yaml
defaults:
  containerResources:
    limits:
      memory:
        value: xxxx
      cpu:
        value: xxxx
    requests:
      memory:
        value: xxxx
      cpu:
        value: xxxx
```

checking if there could be a simpler way to handle this while opening up scope for more defaults.

Reply:
Viper uses the mapstructure package for decoding, which doesn't support resource.Quantity out of the box.

Reply:
updated to cover defaults

Reply:
If Viper does not support, make it a string, and then we parse it.

Reply:
```
data:
  controller-config.yaml: |+
    defaults:
      containerResources: |
          limits:
            cpu: xxxx
            memory: xxxx
          requests:
            cpu: xxx
            memory: xxxx
    isbsvc:
      redis:

Reply:
Actually I didn't get it, Viper does not support customized structs? we are already doing that, right?

Reply:
> Viper does not support customized structs?
No, it does not parse `resource.Quantity` i.e the value of the map object (under limits/resource) correctly - assigns them 0. Which is due to the underlying mapstructure.

Reply:
> If Viper does not support, make it a string, and then we parse it.

Better. updated!